flax/linen/module.py

```diff
@@ -1061,8 +1078,11 @@ def __call__(self, x):
           self.sow('intermediates', 'h', x * 2,
                    init_fn=init_fn, reduce_fn=reduce_fn)
           return x
-      y, state = Foo.apply(params, 1, mutable=['intermediates'])
-      print(state['intermediates'])  # ==> {'h': 3}
+
+      model = Foo2()
+      variables, _ = model.init(jax.random.PRNGKey(0), x).pop('intermediates')
```

Comment:
(note that `variables` is actually empty here...)

Reply:
Thanks, fixed by making it the same as the previous example.